server/server.js

```diff
@@ -91,6 +91,14 @@ export default function server(app: express, options: any):express {
     checkConfiguration({ checkQueryString: true }),
     (req, res) => {
     req.hull.enqueue("startImport");
+
+    if (req.agent) {
+      // Typically the agent closes itself after the query or stream is done
+      // but in this case we're not using it, so we have to close directly
+      // should look into this pattern more
+      req.agent.closeClient();
```

Comment:
Are we opening a connection to the database on those actions ? 
I thought that those routes were only enqueuing the jobs

Reply:
I would not leak the concern of managing the connection to the underlying db in the routing layer of the application...

Reply:
correct.  We instantiate an "agent" in validateMiddleware which injects an agent into the request.  However, in these 2 cases where we just enqueue a job, the agent isn't closed (leaks a client).

I agree the concerns should not be mixed.  I also think we shouldn't be "opening a connection" in the constructor of the agent.  But those are bigger changes that we should maybe think about.  adding this close client prevents leaks in the current pattern.

Reply:
This seems like a bad pattern we should refactor in a version in the future.  Would like to make sure this hotfix is in place first though.

In fact the pattern is even more annoying.  Had to add close client anywhere where the validation middleware existed and did not call a runQuery or startImport